src/helpers/sqlite.ts

```diff
@@ -0,0 +1,191 @@
+import { Expression } from '../expression/expression.js'
+import { AliasNode } from '../operation-node/alias-node.js'
+import { ColumnNode } from '../operation-node/column-node.js'
+import { IdentifierNode } from '../operation-node/identifier-node.js'
+import { ReferenceNode } from '../operation-node/reference-node.js'
+import { SelectQueryNode } from '../operation-node/select-query-node.js'
+import { SelectQueryBuilder } from '../query-builder/select-query-builder.js'
+import { RawBuilder } from '../raw-builder/raw-builder.js'
+import { sql } from '../raw-builder/sql.js'
+import { Simplify } from '../util/type-utils.js'
+
+/**
+ * A SQLite helper for aggregating a subquery into a JSON array.
+ *
+ * NOTE: This helper is only guaranteed to fully work with the built-in `SqliteDialect`.
+ * While the produced SQL is compatible with all SQLite databases, some 3rd party dialects
+ * may not parse the nested results into arrays.
```

Comment:
The built-in `SqliteDialect` doesn't parse the results either. This comment could be replaced with instructions to use the `ParseJSONResultsPlugin`. Same for the other functions in this module.

Reply:
Sounds good 👍 added those notes and instructions